### PR TITLE
PlotFileUtil: Add Direct Includes

### PR DESCRIPTION
**Pull request by ax3l**
## Summary

Adds the used direct includes to `AMReX_PlotFileUtil.H`.

## Additional background

https://github.com/AMReX-Codes/pyamrex/pull/153

## Checklist

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
Correctness here is subjective. I think all those headers have been indirectly included. Are you seeing real issues?

**Comment by ax3l** [comment]
Correctness in the sense of include stability:
This will make it long-term stable if transitive headers change.
https://include-what-you-use.org

Changing the label appropriately to `cleaning`.

**Comment by WeiqunZhang** [comment]
I tried include-what-you-use in the past. It is too much work to make it happy because we have a lot of build options and we need to add a lot of IWYU pragmas. If we were using it, we would have marked AMReX_MultiFab.H as exporting AMReX_Box.H, AMReX_REAL.H etc.

File: Src/Base/AMReX_PlotFileUtil.H [hunk 1]
```diff
@@ -2,14 +2,20 @@
...
 #include <AMReX_Config.H>
 
+#include <AMReX_Array.H>
+#include <AMReX_Box.H>
 #include <AMReX_Geometry.H>
+#include <AMReX_IntVect.H>
 #include <AMReX_MultiFab.H>
 #include <AMReX_PlotFileDataImpl.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Vector.H>
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_PlotFileUtil.H]:
```suggestion
#include <AMReX_Array.H>
#include <AMReX_Geometry.H>
#include <AMReX_MultiFab.H>
#include <AMReX_PlotFileDataImpl.H>
#include <AMReX_Vector.H>
```
It think it's safe to assume they will be "exported" by the MultiFab header.

Comment (WeiqunZhang) [reply Src/Base/AMReX_PlotFileUtil.H]: To be more specific, "they" means Box, IntVect, and Real.

Comment (ax3l) [reply Src/Base/AMReX_PlotFileUtil.H]: Ok :+1:

**Comment by ax3l** [comment]
> I tried include-what-you-use in the past.

Agreed (for the tool). I was more referring to the concept :)

**Comment by WeiqunZhang** [comment]
I agree with the concept too. :)

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**